libraries/ESP8266WiFiMesh/src/ESP8266WiFiMesh.cpp

```diff
@@ -18,151 +18,582 @@
+
+String ESP8266WiFiMesh::last_ssid = "";
+bool ESP8266WiFiMesh::static_IP_activated = false;
+
+// IP needs to be at the same subnet as server gateway (192.168.4 in this case). Station gateway ip must match ip for server.
+IPAddress ESP8266WiFiMesh::static_IP = empty_IP;
+IPAddress ESP8266WiFiMesh::gateway = IPAddress(192,168,4,1);
+IPAddress ESP8266WiFiMesh::subnet_mask = IPAddress(255,255,255,0);
+
+std::vector<NetworkInfo> ESP8266WiFiMesh::connection_queue = {};
+std::vector<TransmissionResult> ESP8266WiFiMesh::latest_transmission_outcomes = {};
+
+ESP8266WiFiMesh::ESP8266WiFiMesh(std::function<String(String, ESP8266WiFiMesh *)> requestHandler, std::function<transmission_status_t(String, ESP8266WiFiMesh *)> responseHandler, 
+                                 std::function<void(int, ESP8266WiFiMesh *)> networkFilter, String mesh_password, String mesh_name, String node_id, bool verbose_mode, uint8 mesh_wifi_channel, int server_port) 
+                                 : _server(server_port), _lwip_version{0, 0, 0}
+{
+  updateNetworkNames(mesh_name, (node_id != "" ? node_id : Uint64ToString(ESP.getChipId())));
+  _requestHandler = requestHandler;
+  _responseHandler = responseHandler;
+  _mesh_wifi_channel = mesh_wifi_channel;
+  _server_port = server_port;
+  _mesh_password = mesh_password;
+  _verbose_mode = verbose_mode;
+  _networkFilter = networkFilter;
+  storeLwipVersion();
+}
+
+void ESP8266WiFiMesh::verboseModePrint(String string_to_print, bool newline)
+{
+  if(_verbose_mode)
+  {
+    if(newline)
+      Serial.println(string_to_print);
+    else
+      Serial.print(string_to_print);
+  }
+}
+
+/**
+ * Note that using a base higher than 16 increases likelihood of randomly generating ssid strings containing controversial words. 
+ * 
+ * @param number The number to convert to a string with radix "base".
+ * @param base The radix to convert "number" into. Must be between 2 and 36.
+ * @returns A string of "number" encoded in radix "base".
+ */
+String ESP8266WiFiMesh::Uint64ToString(uint64_t number, byte base)
+{
+  String result = "";
+
+  while(number > 0)
+  {
+    result = String((uint32_t)(number % base), base) + result;
+    number /= base;
+  }
+
+  return (result == "" ? "0" : result);
+}
+
+/**
+ * Note that using a base higher than 16 increases likelihood of randomly generating ssid strings containing controversial words. 
+ * 
+ * @param string The string to convert to uint64_t. String must use radix "base".
+ * @param base The radix of "string". Must be between 2 and 36.
+ * @returns A uint64_t of the string, using radix "base" during decoding.
+ */
+uint64_t ESP8266WiFiMesh::StringToUint64(String string, byte base)
+{
+  uint64_t result = 0;
+
+  char current_character[1];
+  for(uint32_t i = 0; i < string.length(); i++)
+  {
+    result *= base;
+    current_character[0] = string.charAt(i);
+    result += strtoul(current_character, NULL, base);
+  }
+  
+  return result;
+}
```

Comment:
Can this function fail? what happens if the string passed as arg is invalid or has a mistake?
Does it make sense to put these conversion files elsewhere? At least in standalone files.

Reply:
As far as I can tell, any invalid input is treated as a 0.